ChessBoard.UnitTests/ChessBoardTests.cs

```diff
@@ -0,0 +1,82 @@
+using System.Text;
+using Microsoft.Extensions.Logging;
+using Moq;
+
+namespace ChessBoardApp.ChessBoard.Tests
+{
+    class TestLogger<T> : ILogger<T>
+    {
+        public List<string> LoggedMessages { get; } = new();
+
+        public IDisposable BeginScope<TState>(TState state) => null;
+
+        public bool IsEnabled(LogLevel logLevel) => true;
+
+        public void Log<TState>(
+            LogLevel logLevel,
+            EventId eventId,
+            TState state,
+            Exception exception,
+            Func<TState, Exception, string> formatter
+        )
+        {
+            string logMessage = formatter(state, exception);
+            LoggedMessages.Add(logMessage);
+        }
+    }
+
+    public class ChessBoardGeneratorTests
+    {
+        [Fact]
+        public void GenerateRectangleBoard_ShouldLogCorrectChessBoard()
+        {
+            // Arrange
+            var logger = new TestLogger<ChessBoardGenerator>();
+            var chessBoardGenerator = new ChessBoardGenerator(logger);
+            int width = 4;
+            int length = 6;
+
+            // Act
+            chessBoardGenerator.GenerateRectangeBoard(width, length);
+
+            // Assert
+            string expectedChessBoard = GetExpectedChessBoard(width, length);
+            Assert.Contains(expectedChessBoard, string.Join("\r\n", logger.LoggedMessages));
+        }
+
+        private static string GetExpectedChessBoard(int width, int length)
+        {
+            StringBuilder chessBoard = new();
+            for (int i = 0; i < length; i++)
+            {
+                for (int j = 0; j < width; j++)
+                {
+                    chessBoard.Append((j + i) % 2 == 0 ? "2" : "0");
+                }
+
+                chessBoard.AppendLine();
+            }
+
+            return chessBoard.ToString();
+        }
+    }
+
+    public class ChessBoardDialogTests
```

Comment:
Let's not test dialog as it would become increasingly complicated when testing console without some interface wrapper. We could, on the other hand do some validator tests, for example https://adform.slack.com/archives/C05ECTFN5GE/p1688357426508479?thread_ts=1688335639.652929&cid=C05ECTFN5GE point 6. 
You could provide multiple inputs and assert that ValidationResult is { IsValid == false, ErrorMessage == "validation error message" } and if input is correct - ValidationResult is { IsValid == true, ErrorMessage == null }

Or maybe you have some other suggestions, let's discuss ;)

Reply:
well done!